javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodMapper.java

```diff
@@ -130,6 +131,27 @@ public ProxyMethod map(Operation operation) {
             parameters.add(Mappers.getProxyParameterMapper().map(parameter));
         }
 
+        if (settings.getAddContextParameter()) {
+            ProxyMethodParameter contextParameter = new ProxyMethodParameter(
+                "The context to associate with this operation.",
+                    ClassType.Context,
+                    ClassType.Context,
+                    "context",
+                    RequestParameterLocation.None,
+                    "context",
+                    true,
+                    false,
+                    true,
```

Comment:
Why is it required?

Reply:
I think I've copied it from v3.

Currently the flags is not used, since client code will use FluxUtil.withContext to let user provide context via reactor subscriberContext (though it probably not working nice for PagedFlux).

If we instead decide to let client provide 2 sets of client method (one with context, one without), then we would need isRequred=false here.

Reply:
Just FYI client method parameters are ordered so that required params are asked first, and optional params follow.

Reply:
I see. I've changed it to isRequired=false